### PR TITLE
Use blocks instead of widgets in new WordPress installations

### PR DESCRIPTION
**Pull request by noisysocks**
Modifies `wp_install_defaults()` so that when you install a new WordPress site you have _block_ in your widget areas, not widgets.

Trac ticket: https://core.trac.wordpress.org/ticket/53324

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by noisysocks** [comment]
The Twenty Twenty-one theme will need some styling adjustments to accomodate for the different markup.

**Comment by jasmussen** [comment]
> I could use some guidance from someone who knows all about the default themes. @jasmussen, maybe? 🙂

I _always_ appreciate a ping, thank you. In this case, I can only offer limited guidance:

- A `widget-heading` class would probably be useful, but I imagine it might be onerous to add to the heading block just in this context, and if it's not easy, might not be worth it. 
- I think it's okay for themes to have to do a few updates here and there, and in this case I imagine there's a widget container they can target, i.e. `.widgets h3` instead of `h3.widget-heading`, it shouldn't be a lot of work, and might be worth it for all the other changes  blocks bring to widgets anyway. 
- Finally, the heading block comes with options which classic widget headings didn't. Specifically, you can choose your own heading level, which is an important accessibility consideration for some themes that might be using Widgets in a more primary fashion than others. 
- In addition to being able to set the correct heading level depending on your theme, the heading block comes with the ability to set the font size. That could potentially be a solution here, embracing one of the default smaller presets. I imagine that's why a widget-heading class existed in the first place — H2 might be the correct heading level semantically, but might have been visually too big for the design.

For some actual theme experts, you might want to ping @melchoyce and perhaps @maggiecabrera.

File: src/wp-includes/block-editor.php [hunk 1]
```diff
@@ -245,7 +245,6 @@ function get_default_block_editor_settings() {
 			'media_image',
 			'media_gallery',
 			'media_video',
-			'meta',
```
Comment (draganescu) [inline src/wp-includes/block-editor.php:248]: why is this removed?

File: src/wp-admin/includes/upgrade.php [hunk 2]
```diff
@@ -464,18 +418,18 @@ function wp_install_defaults( $user_id ) {
...
+					0 => 'block-2',
+					1 => 'block-3',
+					2 => 'block-4',
 				),
 				'sidebar-2'           => array(
-					0 => 'archives-2',
-					1 => 'categories-2',
-					2 => 'meta-2',
+					0 => 'block-5',
+					1 => 'block-6',
```
Comment (tellthemachines) [inline src/wp-admin/includes/upgrade.php:427]: Shouldn't we still have a meta widget in here?

File: src/wp-includes/block-editor.php (hunk 1 above)
Comment (tellthemachines) [reply src/wp-includes/block-editor.php:248]: I think that's because we don't have a block with the same functionality.

**Review by tellthemachines — APPROVED** [review]
Code looks good ✅ Tested with both the new and the old screens and it's working well.

File: src/wp-includes/block-editor.php (hunk 1 above)
Comment (noisysocks) [reply src/wp-includes/block-editor.php:248]: Correct. I ended up removing it in https://core.trac.wordpress.org/ticket/53301, though.

File: src/wp-admin/includes/upgrade.php (hunk 2 above)
Comment (noisysocks) [reply src/wp-admin/includes/upgrade.php:427]: I can't decide... would it be weird to have a Legacy Widget block in amongst the default blocks that you see in a new WordPress site? Is it a useful widget to have by default in 2021?

Comment (tellthemachines) [reply src/wp-admin/includes/upgrade.php:427]: Yeah good point about the Legacy Widget. We can leave it out for now and see if anyone notices 😄

**Comment by noisysocks** [comment]
Thanks for your useful comment, @jasmussen!

> A widget-heading class would probably be useful, but I imagine it might be onerous to add to the heading block just in this context, and if it's not easy, might not be worth it.

Yeah. I think adding the `widget-heading` class is not the way to go. It will be confusing to users who insert a new Heading block in their widget area as the new Heading block that they add won't have the class and so will appear bigger.

> Finally, the heading block comes with options which classic widget headings didn't. Specifically, you can choose your own heading level, which is an important accessibility consideration for some themes that might be using Widgets in a more primary fashion than others.

Great point! 🌈 

> In addition to being able to set the correct heading level depending on your theme, the heading block comes with the ability to set the font size. That could potentially be a solution here, embracing one of the default smaller presets. I imagine that's why a widget-heading class existed in the first place — H2 might be the correct heading level semantically, but might have been visually too big for the design.

This is a really good idea and not one I'd thought of. I tried it, but it doesn't work fully because we also want the heading to appear bold. We _could_ do this by adding `<strong>` to the Heading block but I fear that at this point we're having the default blocks assume too much about what everything should look like when really this is the theme's job.

> I think it's okay for themes to have to do a few updates here and there, and in this case I imagine there's a widget container they can target, i.e. .widgets h3 instead of h3.widget-heading, it shouldn't be a lot of work, and might be worth it for all the other changes blocks bring to widgets anyway.

I think this is what we have to do! I've adjust the Twenty Twenty-one styles in 249e2d6. Happy to hear thoughts from everyone.

**Comment by noisysocks** [comment]
Unfortunately this breaks just about every unit test we have 😛. I'll start fixing them on Monday.

File: src/wp-admin/includes/upgrade.php (hunk 2 above)
Comment (hellofromtonya) [reply src/wp-admin/includes/upgrade.php:427]: I think it makes sense to not have this legacy widget on a new install. Why? The default experience of a new install does not include the Meta widget in an active sidebar.

That said, for those who want to use its features, might be good to have a new block version of it.

Comment (noisysocks) [reply src/wp-admin/includes/upgrade.php:427]: That makes sense to me. I've kept it removed.

Comment (draganescu) [reply src/wp-admin/includes/upgrade.php:427]: My worry here is that we assume it's not useful. Isn't the meta widget the default front end UX path to wp-admin?

Comment (critterverse) [reply src/wp-admin/includes/upgrade.php:427]: I would vote for leaving the Meta widget out but would it be possible for it to be available in the inserter and in the Legacy Widget block dropdown list until a new block version is created? (Similar to what we’re looking at doing for the Navigation Menu widget in [#32431](https://github.com/WordPress/gutenberg/pull/32431)?)

Comment (noisysocks) [reply src/wp-admin/includes/upgrade.php:427]: > My worry here is that we assume it's not useful. Isn't the meta widget the default front end UX path to wp-admin?

It already does not show on a new WordPress website because Twenty Twenty-one only has one widget area.

> I would vote for leaving the Meta widget out but would it be possible for it to be available in the inserter and in the Legacy Widget block dropdown list until a new block version is created? (Similar to what we’re looking at doing for the Navigation Menu widget in #32431?)

Yes absolutely. Would you mind creating an issue in the Gutenberg repo about adding a block equivalent of Meta? I looked but couldn't find one.

Comment (critterverse) [reply src/wp-admin/includes/upgrade.php:427]: No prob, made an issue for the block version here: https://github.com/WordPress/gutenberg/issues/32551
